src/grpc_requests/client.py

```diff
@@ -383,6 +386,13 @@ def _get_file_descriptor_by_symbol(self, symbol):
 
     def _register_file_descriptor(self, file_descriptor):
         logger.debug(f"start {file_descriptor.name} register")
+        try:
+            self._desc_pool.FindFileByName(file_descriptor.name)
+        except KeyError:
+            pass
+        else:
+            logger.debug(f'{file_descriptor.name} already registered')
+            return
```

Comment:
```suggestion
        if file_descriptor.name in self.registered_file_names:
           logger.debug(f'{file_descriptor.name} already registered')
           return
```

Do you think that you could leverage the set of registered file names to accomplish this as well?

I also bet there's a way to refactor the recursive loop to lean more properly on the methods provided by the DescriptionPool the way you're doing here.

Reply:
@ViridianForge gave it a shot. See the latest commit

Reply:
Forgot to do the async side. Give me a minute

Reply:
Done